src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/web/ReportRunController.kt

```diff
@@ -1,26 +1,42 @@
 package org.vaccineimpact.orderlyweb.controllers.web
 
 import org.vaccineimpact.orderlyweb.ActionContext
+import org.vaccineimpact.orderlyweb.OrderlyServer
+import org.vaccineimpact.orderlyweb.OrderlyServerAPI
 import org.vaccineimpact.orderlyweb.controllers.Controller
+import org.vaccineimpact.orderlyweb.db.AppConfig
 import org.vaccineimpact.orderlyweb.db.repositories.OrderlyWebReportRunRepository
 import org.vaccineimpact.orderlyweb.db.repositories.ReportRunRepository
 import org.vaccineimpact.orderlyweb.models.ReportRunLog
+import org.vaccineimpact.orderlyweb.models.ReportStatus
 import org.vaccineimpact.orderlyweb.models.ReportRunWithDate
 
 class ReportRunController(
     context: ActionContext,
-    private val reportRunRepository: ReportRunRepository
+    private val reportRunRepository: ReportRunRepository,
+    private val orderlyServerAPI: OrderlyServerAPI
 ) : Controller(context)
 {
     constructor(context: ActionContext) : this(
             context,
-            OrderlyWebReportRunRepository()
+            OrderlyWebReportRunRepository(),
+            OrderlyServer(AppConfig()).throwOnError()
     )
 
     fun getRunningReportLogs(): ReportRunLog
     {
         val key = context.params(":key")
-        return reportRunRepository.getReportRun(key)
+        var log = reportRunRepository.getReportRun(key)
+        if (log.status in listOf(null, "queued", "running"))
+        {
+            val statusResponse = orderlyServerAPI.get(
+                    "/v1/reports/$key/status/",
+                    mapOf("output" to "true"))
+            val latestStatus = statusResponse.data(ReportStatus::class.java)
+            reportRunRepository.updateReportRun(key, latestStatus.status, latestStatus.version, latestStatus.output)
```

Comment:
`reportRunRepository.updateReportRun` should update the table but what if there was a problem ?

Reply:
If there's an exception in the updateReportRun method (eg unknown key) that result in an error response which should be displayed in the front end

Reply:
updateReportRun does't have a throwable at the moment.

Reply:
Actually, having said that, it's a SQL update so I it wouldn't throw an exception if the key was unknown in the updateReportRun, but it would have already thrown one when it didn't find the key on the initial getReportRun. So I think we're still ok!